### PR TITLE
(improvement) Update Electron to v17

### PR DESCRIPTION
**Pull request by Blaumaus**
ASANA Ticket: [Upgrade electron to the latest version](https://app.asana.com/0/1125859137800433/1201766611631274/f)

Changelog:
 - updated Electron to v17
 - (minor) updated other dependencies to latest version

UI Demonstration:
Screenshot from 2022-02-07 14-38-15

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,10 +112,10 @@
   ],
```
Comment (tarcisiozf) [inline package.json:112]: can we update the [engine](https://github.com/bitfinexcom/bfx-hf-ui/blob/master/package.json#L6) to `>=16`? Ref: https://www.electronjs.org/blog/electron-17-0#stack-changes

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [12.x]
+        node-version: [16.x]
```
Comment (bhoomij) [inline .github/workflows/pull_request.yml:21]: lets set this to node 14 as electron build server only supports node 14.x (https://bitfinex.slack.com/archives/C02UVLC5ZD0/p1644227918521639?thread_ts=1643914764.588169&cid=C02UVLC5ZD0)

File: package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "version": "3.26.0",
   "description": "Official bitfinex honey framework UI - for live trading and executing algorithmic orders/strategies",
   "engines": {
-    "node": ">=6"
+    "node": ">=16"
```
Comment (bhoomij) [inline package.json:6]: ^ node 14

**Review by bhoomij — CHANGES_REQUESTED** [review]

**Comment by bhoomij** [comment]
@Blaumaus could you share macos binaries either here or via slack ? just want to try and make sure electron 17 works with node 14 too!

File: package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "version": "3.26.0",
   "description": "Official bitfinex honey framework UI - for live trading and executing algorithmic orders/strategies",
   "engines": {
-    "node": ">=6"
+    "node": ">=14"
```
Comment (tarcisiozf) [inline package.json:6]: please change the version on the [.tool-versions](https://github.com/bitfinexcom/bfx-hf-ui/blob/master/.tool-versions)

**Review by tarcisiozf — CHANGES_REQUESTED** [review]

**Comment by bhoomij** [comment]
FYI: I am facing issue on macOS with this PR, holding merge until I could spent more time debug and fix for mac

**Pull request closed without merging**